!excellence Suport for TUM Bot

CommanderStorm opened the pull request:
The Bot defenitively needs some !excellence

CommanderStorm committed:
Added the Excellence.txt file for !excellence support

CommanderStorm committed:
Added Excellence.py Cog for !excellence support

CommanderStorm committed:
fixed a typo

NiOHar commented:
It's quite an excellent feature ;)
Defo nice to have.

timschumi commented on exzellenz.txt:
```diff
@@ -0,0 +1,14 @@
+151 Jahre Exzellenz.
+Die TUM ist sehr exzellent.
+PgDP ist sehr exzellent.
```
PGdP

timschumi commented on exzellenz.txt:
```diff
@@ -0,0 +1,14 @@
+151 Jahre Exzellenz.
+Die TUM ist sehr exzellent.
+PgDP ist sehr exzellent.
+EiDI ist sehr exzellent.
```
EidI

timschumi commented on exzellenz.txt:
```diff
@@ -0,0 +1,14 @@
+DS ist sehr exzellent.
+ERA ist sehr exzellent.
+Die Mensa ist sehr exzellent.
+Die Bockabfertigung in der Mensa ist sehr exzellent.
```
Bockabfertigung?

CommanderStorm replied:
Ja, Blockabfertigung.
https://mensa.liste.party/
https://www.sueddeutsche.de/muenchen/landkreismuenchen/garching-tum-mensa-eroeffnung-ansturm-1.4661339

timschumi commented on exzellenz.txt:
```diff
@@ -0,0 +1,14 @@
+Die Bockabfertigung in der Mensa ist sehr exzellent.
+Die Vorlesungssäle im MW sind sehr exzellent.
+DS ist mir zu trivial.
+PgDP ist mir zu trivial.
```
PGdP

timschumi commented on exzellenz.txt:
```diff
@@ -0,0 +1,14 @@
+Die Vorlesungssäle im MW sind sehr exzellent.
+DS ist mir zu trivial.
+PgDP ist mir zu trivial.
+EiDI ist mir zu trivial.
```
EidI

timschumi reviewed (changes requested):
Maybe somehow make this use the same codebase as the Quotes thing that this is based on?
Another option would be to generate the message on-the-fly (combination of subject + exzellent/trivial).

Also, I'm not sure about the usefulness of this in general, I don't really think it's a funny addition at all.

CommanderStorm committed:
added excellence and trivial autocomplete messages. Now Just the first Part has to be added to the .txt file.
Also renamed the quotes.py file and merged the excellenz.py file into the new UserGeneratedContentFile

CommanderStorm committed:
fixed the Typos

CommanderStorm committed:
removed unnessesary variable calling

CommanderStorm committed:
added a bit more Content

CommanderStorm commented:
I think the separation of the "UserGeneratedContent" (I know the name is not that good, but I cant think of a more fitting one) into its own folder was nessesary, as now there is more than one file and because this cleans things up a notch..

CommanderStorm commented:
different files for trivial/excellence were neccesary, as some of the things dont really make sense in the other context. (i.e. "Die Mensa ist trivial")

timschumi commented:
I'm not really convinced by the code quality yet, I'll see if I can make a few changes in the next days.

timschumi committed:
Fix typo

timschumi committed:
Rename UserGeneratedContent to "strings"

timschumi committed:
Merge UserGeneratedContent classes into Randomstuff

timschumi committed:
Introduce a randomstring() Method

timschumi committed:
Rework file structure and algorithm

timschumi committed:
Integrate trivial_autocomplete into extra strings

timschumi commented:
I fixed some of the major issues I had with this. However, some of the strings are still nonsensical, so please redistribute them across the two lists (`exzellenz_extra` are taken as is, `exzellenz_trivial` is extended with either "ist sehr exzellent" or "ist trivial").

If the result uses more than those two lists, I'll just reject and lock this Pull Request, because this should not take that much effort to be maintained, and I don't feel like importing a module that requires this much effort to be maintained, seeing how this is neither useful nor funny.

CommanderStorm committed:
fixed a missing bracket

CommanderStorm committed:
redistribution of strings

CommanderStorm committed:
added Fun-Fact

CommanderStorm committed:
added another Fun-Fact
